src/node/node_state.h

```diff
@@ -2265,20 +2271,30 @@ namespace ccf
             return std::make_unique<ResharingsHook>(version, w);
           }));
 
-      network.tables->set_map_hook(
+      // Note: The Signatures hook and SerialisedMerkleTree hook are separate
+      // because the signature and the Merkle tree are recorded in distinct
+      // tables (for serialisation performance reasons). However here, they are
+      // expected to always be called together and for the same version as they
+      // are always written by each signature transaction.
+
+      network.tables->set_global_hook(
```

Comment:
This should be `set_map_hook()` and return `return kv::ConsensusHookPtr(nullptr);`.